src/BotwinExtensions.cs

```diff
@@ -44,41 +45,52 @@ public static IApplicationBuilder UseBotwin(this IApplicationBuilder builder, Bo
                 {
                     var moduleType = module.GetType();
 
-                    foreach (var route in module.Routes)
+                    foreach (var route in module.Routes.Keys)
                     {
-                        routeBuilder.MapVerb(route.verb, route.path, ctx =>
-                        {
-                            var statusCodeHandlers = ctx.RequestServices.GetServices<IStatusCodeHandler>();
-
-                            var requestScopedModule = ctx.RequestServices.GetRequiredService(moduleType) as BotwinModule;
-
-                            // TODO: Use the handler from the resolved 'requestScopedModule' instead of the one from the "startup scope".
-                            var routeHandler = requestScopedModule.Routes.FirstOrDefault(x => x.path == route.path && x.verb == route.verb).handler;
-                            var handler = CreateModuleBeforeAfterHandler(requestScopedModule, routeHandler);
-
-                            var finalHandler = CreateFinalHandler(handler, statusCodeHandlers);
-
-                            return finalHandler.Invoke(ctx);
-                        });
+                        routeBuilder.MapVerb(route.verb, route.path, CreateRouteHandler(route, moduleType));
                     }
                 }
             }
 
             return builder.UseRouter(routeBuilder.Build());
         }
 
-        private static RequestDelegate CreateFinalHandler(RequestDelegate handler, IEnumerable<IStatusCodeHandler> statusCodeHandlers)
+        private static RequestDelegate CreateRouteHandler((string verb, string path) route, Type moduleType)
         {
-            RequestDelegate finalHandler = async (ctx) =>
+            return async (HttpContext ctx) =>
             {
+                var module = ctx.RequestServices.GetRequiredService(moduleType) as BotwinModule;
+                if (!module.Routes.TryGetValue((route.verb, route.path), out var routeHandler))
```

Comment:
I guess this makes both verb (method) and path case sensitive? Is that OK?

Reply:
Yeah, you’re right. If that’s not desired, we should pass a different `IEqualityComparer` to the dictionary constructor in `BotwinModule`. That way, it would still keep the lookup benefits.

For string keys, we could use the built-in `StringComparer.InvariantCultureIgnoreCase`. For the two-tuple key, I guess we will have to run our own comparer, e.g. as shown [here](https://stackoverflow.com/questions/16428492/case-insensitive-dictionary-with-tuple-key). So it would probably look like this:

```
class TwoTupleStringComparer : IEqualityComparer<ValueTuple<string, string>>
{
    public bool Equals(ValueTuple<string, string> x, ValueTuple<string, string> y)
        => StringComparer.InvariantCultureIgnoreCase.Equals(x.Item1, y.Item1) && StringComparer.InvariantCultureIgnoreCase.Equals(x.Item2, y.Item2);

    public int GetHashCode(ValueTuple<string, string> obj)
        => (StringComparer.InvariantCultureIgnoreCase.GetHashCode(obj.Item1), StringComparer.InvariantCultureIgnoreCase.GetHashCode(obj.Item2)).GetHashCode();
}
```

Reply:
I'm not sure if using case-insensitive comparison for the path would be correct. What does Botwin do already? I guess the old check actually _was_ case-senstitive (`==`).